apps/src/code-studio/progressRedux.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * Reducer and actions for progress
+ */
+
+import { SUBMITTED_RESULT, mergeActivityResult, activityCssClass } from './activityUtils';
+
+// Action types
+const INIT_PROGRESS = 'progress/INIT_PROGRESS';
+const MERGE_PROGRESS = 'progress/MERGE_PROGRESS';
+const UPDATE_FOCUS_AREAS = 'progress/UPDATE_FOCUS_AREAS';
+const SHOW_LESSON_PLANS = 'progress/SHOW_LESSON_PLANS';
+
+const initialState = {
+  currentLevelId: null,
+  professionalLearningCourse: null,
+  // a mapping of level id to result
+  progress: {},
+  focusAreaPositions: [],
+  saveAnswersBeforeNavigation: null,
+  stages: null
+};
```

Comment:
TODO - currently passing in initialState as part of createStore. Rationalize these two